packages/datastore-storage-adapter/__tests__/ExpoSQLiteDatabase.test.ts

```diff
@@ -0,0 +1,535 @@
+import ExpoSQLiteDatabase from '../src/ExpoSQLiteAdapter/ExpoSQLiteDatabase';
+import { ParameterizedStatement } from '../src/common/types';
+
+jest.mock('expo-sqlite', () => ({
+	openDatabase: jest.fn(() => mockDatabase),
+}));
+
+jest.mock('expo-file-system', () => ({
+	deleteAsync: jest.fn(path => null),
+	documentDirectory: '/sample/path',
+}));
+
+const mockDatabase = {
+	readTransaction: jest.fn(),
+	transaction: jest.fn(),
+	_db: { close: () => {} },
+};
+const mockTransaction = { executeSql: jest.fn() };
+
+const MOCK_SQL_ERROR = {
+	code: 400,
+	message: 'Error performing the transaction',
+};
+
+const DEFAULT_STATEMENT = 'SELECT * FROM ABCD';
+const DEFAULT_STATEMENT_PARAMS = [];
+const MOCK_RESULT_ARRAY_NUMBER = [1, 2, 3, 4];
+const MOCK_RESULT_ARRAY_STRING = ['this', 'is', 'a', 'string'];
+
+const MOCK_RESULT_SET_NUMBER = {
+	rows: {
+		_array: MOCK_RESULT_ARRAY_NUMBER,
+	},
+};
+
+const MOCK_RESULT_SET_STRING = {
+	rows: {
+		_array: MOCK_RESULT_ARRAY_STRING,
+	},
+};
+
+const MOCK_RESULT_SET_EMPTY = {
+	rows: {
+		_array: [],
+	},
+};
+
+describe('ExpoSQLiteDatabase', () => {
+	const instance = new ExpoSQLiteDatabase();
+	mockDatabase.readTransaction.mockImplementation(func => {
+		func(mockTransaction);
+	});
+	mockDatabase.transaction.mockImplementation(func => {
+		func(mockTransaction);
+	});
```

Comment:
This is kind of nit but I think we can actually move these up into the declarations above since we don't reset them e.g.:
```js
const mockDatabase = {
	readTransaction: jest.fn(func => {
		func(mockTransaction);
	}),
	transaction: jest.fn(func => {
		func(mockTransaction);
	}),
```

Reply:
Makes sense. I will move it upwards.